benchmark/performance_utils.py

```diff
@@ -14,34 +14,50 @@
 
 class Benchmark:
     def __init__(
-        self, op_name, torch_op, arg_func, dtypes, batch, sizes, kwargs_func=None
+        self,
+        op_name,
+        torch_op,
+        arg_func,
+        dtypes,
+        batch,
+        sizes,
+        is_backward=False,
+        kwargs_func=None,
     ):
         self.op_name = op_name
+        if is_backward:
+            self.op_name += " backward"
         self.torch_op = torch_op
         self.arg_func = arg_func
         self.kwargs_func = kwargs_func
         self.dtypes = dtypes
         self.batch = batch
         self.sizes = sizes
         self.gems_op = None
+        self.is_backward = is_backward
 
     def set_gems(self, gems_op):
         self.gems_op = gems_op
 
     def profile(self, op, *args, **kwargs):
+        fn = lambda: op(*args, **kwargs)
+        if self.is_backward:
+            out = fn()
+            dout = torch.randn_like(out)
+            fn = lambda: out.backward(dout, retain_graph=True)
         if CPU_MODE:
             for i in range(WARMUP):
-                op(*args, **kwargs)
+                fn()
             torch.cuda.synchronize()
             start = time.time()
             for i in range(REPETITION):
-                op(*args, **kwargs)
+                fn()
             torch.cuda.synchronize()
             end = time.time()
             latency = (end - start) / REPETITION * 1000
         else:
             latency = triton.testing.do_bench(
-                lambda: op(*args, **kwargs),
+                fn(),
```

Comment:
Maybe it should be fn ? What's the triton version in your testing environment?

Reply:
ok. I have changed fn() to fn. 
My trion version is 3.0.0.